src/tests/test_e2e_install.py

```diff
@@ -29,3 +37,86 @@ def test_networking(self, cluster, network_type):
     def test_olm_operator(self, cluster, olm_operators):
         cluster.prepare_for_installation()
         cluster.start_install_and_wait_for_installed()
+
+    @JunitTestSuite()
+    def test_upgrade_agent(self, cluster):
+        """
+        This test will prepare the cluster with the default agent image. Once it read to install it will restart the
+        service with an agent image that doesn't exist, so that all the hosts will move to `insufficient`. Then it will
+        restart the service again with the original image, so that host will upgrade and move to `ready` again.
+        """
+        # Find the namespace:
+        namespace = os.environ.get("NAMESPACE", "assisted-installer")
+
+        # Create the API client:
+        client = kubernetes.dynamic.DynamicClient(
+            kubernetes.client.api_client.ApiClient(
+                configuration=kubernetes.config.load_kube_config(),
+            ),
+        )
+        configmaps_api = client.resources.get(api_version="v1", kind="ConfigMap")
```

Comment:
if it might affect other tests running in parallel, we might need a way to separate it from other runs and mark it as something that needs to run on its own
is it such a thing?

Reply:
This will indeed affect other tests that run in parallel if they use the same instance of the service, as it will be restarted with an alternative (and sometimes broken) agent image. How do we do that separation? A new test class? Something else?

Reply:
@eliorerz can you advice?

Reply:
Maybe we can generate a new namespace the same we are doing on the kube_api flow:
https://github.com/openshift/assisted-test-infra/blob/master/src/tests/base_kubeapi_test.py#L39-#L41

Reply:
> A new test class
as a start, we should definitely move it away from ``test_e2e_install.py``, which is meant to have more like 'happy-flow' cases. Maybe it's even enough

As for the namespace idea, is it possible? Doesn't it require installing assisted on that namespace? (which we usually are doing outside of the scope of pytest tests)

Reply:
Moved the test to a new `TestUpgradeAgent`class.

Doing a separate installation of the service seems complicated to do from the test itself. Can we instead have a separate job that only runs the tests inside this new `TestUpgradeAgent` class?

Reply:
I'm actually wondering if we even want to test this upgrade feature periodically
The reason I'm not sure about that is because it's something that happens in between version upgrades (and not an installation method that happens regularly day-to-day) and mainly important in the presence of incompatibility between agent behavior (which is also quite rare I think)

Not to say the feature is not important (because it is!). If test costs weren't high I wouldn't mind adding much more tests than today

Reply:
as for this PR itself, it's okay as-is I think. just having it in a separate file and making sure not to run it in a bigger suite will be enough

Reply:
and of course making the job optional costs us nothing (forgot to say that before)

Reply:
That sounds reasonable to me @osherdp. How do we make the job optional?

Reply:
Crystal is doing something similar in her new flow: https://github.com/openshift/release/pull/32302
The important bits are ``always_run: false`` and ``optional: true``

And feel free to add it optionally to all relevant component repos (assisted-test-infra, assisted-service, assisted-installer-agent, etc. whatever code that might affect the behavior)

Reply:
Can you add those jobs before merging this PR? So we can be assured it's passing successfully right from the start on the CI system

Reply:
@osherdp I think this test is more fit to be in "QE tests pool", WDYT?